### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by samlorlah**
- Add Redux Toolkit(`npm install react-redux @reduxjs/toolkit`).
- Structure your application files using a "feature folder" approach and use the *ducks* pattern for your Redux files.
- Write your book's actions and reducer (in `/src/redux/books/books.js`) using the *ducks* pattern. 
    - Define *action types* for adding and removing a book.
    - Set the initial state to be an empty array of books.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - Define state changes for the actions that you created.
        - In case of unknown action - return the current state.
- Write your categories actions and reducer (in `/src/redux/categories/categories.js`) using the *ducks* pattern. 
    - Define an *action type* for checking the status.
    - Set the initial state to be an empty array of categories.
    - Export *Action Creators* for your *actions*.
    - Write your reducer and export it as default.
        - For the check status action return a string "Under construction".
        - In case of unknown action - return the current state.

- Configure the Redux Store (`/src/redux/configureStore.js`):
    - Importing the necessary methods from Redux Toolkit.
    - Combine both reducers into a root reducer by using  `configureStore` function.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Actions
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
```
Comment (TedLivist) [inline src/redux/books/books.js]: - [ ] Consider using [the duck pattern](https://github.com/erikras/ducks-modular-redux) for your action types definition.

  ```
  const ADD_BOOK = 'bookstore/books/ADD_BOOK';
  const REMOVE_BOOK = 'bookstore/books/REMOVE_BOOK';
  ```

File: src/redux/books/books.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  switch (action.type) {
+    default:
+      return state;
+  }
+};
+
+export const addBook = () => ({ type: ADD_BOOK });
+
+export const removeBook = () => ({ type: REMOVE_BOOK });
+
```
Comment (TedLivist) [inline src/redux/books/books.js]: - [x] From the project you can see we are trying to add and remove books. That means apart from the action types, you still need to include the payloads that would be used to work on the store.

  ```
  export const addBook = (payload) => ({
    type: ADD_BOOK,
    payload,
  });
  
  export const removeBook = (payload) => ({
    type: REMOVE_BOOK,
    id: payload,
  });
  ```

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// Actions
+const ADD_BOOK = 'ADD_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
+const booksReducers = (state = [], action) => {
+  switch (action.type) {
+    default:
+      return state;
+  }
+};
```
Comment (TedLivist) [inline src/redux/books/books.js:10]: - [x] Your books reducer still needs to implement the actual addition and removal of books. Right now it just returns the default state.

  ```
  const booksReducers = (state = [], action) => {
    switch (action.type) {
      case ADD_BOOK:
        return [
          action.payload,
          ...state,
        ];
      case REMOVE_BOOK:
        return state.filter((book) => book.id !== action.id);
  
      default:
        return state;
    }
  };
  ```

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Actions
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (TedLivist) [inline src/redux/categories/categories.js]: - [x] Kindly use the ducks' pattern here as well

  ```
  const CHECK_STATUS = 'bookstore/categories/CHECK_STATUS';
  ```

File: src/redux/categories/categories.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+// Reducer
+const categoriesReducers = (state = [], action) => {
+  switch (action.type) {
+    case CHECK_STATUS:
+      return ['Under Contruction'];
```
Comment (TedLivist) [inline src/redux/categories/categories.js]: - [x] We are expected to return the string 'Under construction', not an array. Kindly fix that.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @samlorlah,

Good job so far!
✔️ Good documentation
✔️ Implementation of store config
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 2 above)
Comment (samlorlah) [reply src/redux/books/books.js]: Resolved

File: src/redux/books/books.js (hunk 3 above)
Comment (samlorlah) [reply src/redux/books/books.js:10]: Resolved

File: src/redux/categories/categories.js (hunk 4 above)
Comment (samlorlah) [reply src/redux/categories/categories.js]: Resolved

File: src/redux/categories/categories.js (hunk 5 above)
Comment (samlorlah) [reply src/redux/categories/categories.js]: Resolved

**Review by rustam-tolipov — APPROVED** [review]
## Hello @samlorlah 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**